pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -451,7 +451,16 @@ func (f *ring0Factory) Resumer(info *resource.Info) ([]byte, error) {
 }
 
 func (f *ring0Factory) DefaultNamespace() (string, bool, error) {
-	return f.clientConfig.Namespace()
+	namespace, boolRetval, err := f.clientConfig.Namespace()
```

Comment:
My intent was just to allow it to be valid in `-n`. Would it be an acceptable compromise to throw a fatal error inside here if the result of the trim operation ends up being empty, since that's the only error condition?